src/impl.cpp

```diff
@@ -1 +1,37 @@
 #include "impl.h"
+
+void Decoder(const char key[],char message[])
+{
+    int ii = 0, jj = 0, n = 0, k = 0;
```

Comment:
**please don't declare more than one variable in one line!
use descriptive names**